src/App/Serialization/JsonRpcRequestDenormalizer.php

```diff
@@ -50,11 +50,7 @@ protected function bindIdIfProvided(JsonRpcRequest $request, array $item) : void
     {
         /** If no id defined => request is a notification */
         if (isset($item[self::KEY_ID])) {
-            $request->setId(
-                $item[self::KEY_ID] == (string)((int)$item[self::KEY_ID])
-                    ? (int)$item[self::KEY_ID] // Convert it in case it's a string containing an int
```

Comment:
Unsure why I did that 🤔 Saw nothing on JSON-RPC specification regarding that point => removed